tools/ds-identify

```diff
@@ -1579,6 +1581,10 @@ collect_info() {
 }
 
 print_info() {
+    DI_LOG=stderr
```

Comment:
my understanding is that this is correct
although it didn't seem to be necessary before

Reply:
Why do we need the `DI_LOG=stderr` and `ensure_sane_path` calls?

I think that I see why `read_uname_info` and `set_run_path` might be required - this function is the other "main" entry point of this function, thus will require the `PATH_RUN*` variables to be properly set, as they were before this function.

Reply:
here's what happens without `DI_LOG=stderr`:

```
tools/ds-identify: 185: 3: Bad file descriptor
tools/ds-identify: 185: 3: Bad file descriptor
```

Reply:
but `set_run_path` is not, so i removed that.

Reply:
I meant the other way around:
```diff
diff --git a/tools/ds-identify b/tools/ds-identify
index 6e923a550..154668a00 100755
--- a/tools/ds-identify
+++ b/tools/ds-identify
@@ -1584,8 +1584,7 @@ collect_info() {
 }
 
 print_info() {
-    DI_LOG=stderr
-    ensure_sane_path
+    set_run_path
     read_uname_info
     collect_info
     _print_info
```

since `set_run_path` sets DI_LOG, we shouldn't unconditionally log to stderr, in case of `DI_MAIN=print_info DI_LOG=/some/file tools/ds-identify` or similar.

The above didn't have the bad descriptor you mentioned.